### PR TITLE
Add validation to requestCall page and add policy check

### PR DESCRIPTION
**Pull request by parasharrajat**
### Details

https://github.com/Expensify/App/issues/4861#issuecomment-906962896
### Fixed Issues

$ https://github.com/Expensify/App/issues/4861

### Tests | QA Steps

   1. Go to staging.new.expensify.com
   2. Login with any account
   3. Search for Concierge
   4. Click on the Call icon
   5. Don't write anything on the phone number and click call.
   6. Nothing should happen.
   7. Enter a phone number and click call.

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web | Mobile Web | Desktop

image
image

#### Android

Screenshot 2021-09-10 00:52:18

### CODE REVIEW

**Comment by parasharrajat** [comment]
@Jag96 Please let me know if you want to add the Phone number validation on the frontend as well.

**Comment by Jag96** [comment]
@parasharrajat yes I think we should add the phone number validation you mentioned in your proposal to this as well 👍

**Comment by parasharrajat** [comment]
@Jag96 one Question: 
What should I do?
1. I can disable the button until the phone is valid.
2. I can let the button enabled for invalid phone and validate the number on submit and then show red border around the input with a message `Please provide a valid phone number.`
3. I can show the red border as soon as the user starts typing and shown, until the entered number is not correct. I don't feel this is right as another part of the app does not follow this.

**Comment by Jag96** [comment]
Good question, commented in [slack](https://expensify.slack.com/archives/C01GTK53T8Q/p1631218542139500) to see what the best way to handle this would be.

**Comment by parasharrajat** [comment]
@Jag96 Updated. Please let me know the error message and its translation in Spanish.

**Comment by Jag96** [comment]
@parasharrajat let's use this one: https://github.com/Expensify/App/blob/dad9ddd4b1bf280510322329f37ee66302df6126/src/languages/en.js#L636

We no longer show it in a growl, so maybe update the name of it to `errorMessageInvalidPhone` instead

**Comment by parasharrajat** [comment]
Sure. thanks

**Comment by parasharrajat** [comment]
Updated.

**Review by Jag96 — COMMENTED** [review]
Based on the updates in [slack](https://expensify.slack.com/archives/C01GTK53T8Q/p1631303572177600?thread_ts=1631218542.139500&cid=C01GTK53T8Q) can you update this to use the solution in step 2 of https://github.com/Expensify/App/pull/5169#issuecomment-916388582 so the submit button isn't disabled?

**Comment by parasharrajat** [comment]
Ok. 
Now, How should we manage the behavior?
1. Button is always active.
2. User enters wrong phone, we show `That doesn’t look like a valid phone number. Try again with the country code. e.g. +15005550006` error.
3. But what if user tries to submit without any phone number, Will the error be shown? if so, what will be the error message?
4. There are first and last name fields, Growl is shown when any of them is empty. Is this the correct behavior?

**Comment by Jag96** [comment]
> Button is always active.

👍 

> User enters wrong phone, we show That doesn’t look like a valid phone number. Try again with the country code. e.g. +15005550006 error.

👍 

> But what if user tries to submit without any phone number, Will the error be shown? if so, what will be the error message?

We can show an error and use https://github.com/Expensify/App/blob/e33fcf7c049b92f3aac2b7c8d91e54b9bed584b8/src/languages/en.js#L428 if the phone number is blank on submit.

> There are first and last name fields, Growl is shown when any of them is empty. Is this the correct behavior?

Yes, that's using a different component so I'd leave that functionality the same for this PR. It looks like the growl is using `success` though (`Growl.success(this.props.translate('requestCallPage.growlMessageEmptyName'))`), can you update it to use `error` instead?

**Comment by parasharrajat** [comment]
@Jag96 Updated

File: src/pages/RequestCallPage.js [hunk 1]
```diff
@@ -184,7 +195,15 @@ class RequestCallPage extends Component {
...
-                                onChangeText={phoneNumber => this.setState({phoneNumber})}
+                                errorText={this.state.phoneNumberError}
+                                onBlur={() => {
+                                    if (_.isEmpty(this.state.phoneNumber.trim())) {
+                                        this.setState({phoneNumberError: this.props.translate('messages.noPhoneNumber')});
+                                    } else if (!Str.isValidPhone(this.state.phoneNumber)) {
+                                        this.setState({phoneNumberError: this.props.translate('requestCallPage.errorMessageInvalidPhone')});
+                                    }
+                                }}
+                                onChangeText={phoneNumber => this.setState({phoneNumber, phoneNumberError: ''})}
```
Comment (Jag96) [inline src/pages/RequestCallPage.js:206]: It looks like clearing the phoneNumberError here is causing the error message to disappear as soon as the user starts typing, so the user can't see the phone number example as they type. Can you set the phoneNumberError to empty inside the onSubmit function instead so the error message remains until they resubmit?

**Review by Jag96 — CHANGES_REQUESTED** [review]

File: src/pages/RequestCallPage.js (hunk 1 above)
Comment (parasharrajat) [reply src/pages/RequestCallPage.js:206]: Yeah

**Comment by parasharrajat** [comment]
Done

**Review by Jag96 — APPROVED** [review]
LGTM, thanks for updating!

**Comment by isagoico** [comment]
@parasharrajat @Jag96 is this change behind a beta? This is working for us in expensifail accounts **only**. Gmail accounts are giving us an error and the PR is a fail.

**Comment by parasharrajat** [comment]
I was using private email and I will test using Gmail email.

**Comment by isagoico** [comment]
@parasharrajat Should these tests pass with a gmail too? or is it expected that an error is shown there?

**Comment by parasharrajat** [comment]
@isagoico For me, I don't see any error till step 6. 

On 7, I see.
 
Screenshot 2021-09-14 21:35:54

**Comment by Jag96** [comment]
@isagoico the free plan is still under beta, so if the user isn't on the beta they will see this error message. Gmail users won't be on the beta so it is normal for them to see this error.

**Comment by isagoico** [comment]
Ohh ok got it. Thanks! Looks like we cab check it off then 🎉

**Comment by botify** [comment]
This has been deployed to production and is now subject to a 7-day regression period.
If no regressions arise, payment will be issued on 2021-09-22.  :confetti_ball:

**Pull request merged**